lib/fencing/st_client.c

```diff
@@ -1423,6 +1424,19 @@ xml_to_event(xmlNode *msg)
             event->client_origin = crm_element_value_copy(data, F_STONITH_CLIENTNAME);
             event->device = crm_element_value_copy(data, F_STONITH_DEVICE);
         }
+
+    } else if (pcmk__str_any_of(ntype,
+                                STONITH_OP_DEVICE_ADD, STONITH_OP_DEVICE_DEL,
+                                STONITH_OP_LEVEL_ADD, STONITH_OP_LEVEL_DEL,
+                                NULL)) {
+        xmlNode *data = get_event_data_xml(msg, ntype);
+
+        if (data == NULL) {
+            crm_err("No data for %s event", ntype);
+            crm_log_xml_notice(msg, "BadEvent");
+        } else {
+            event->device = crm_element_value_copy(data, F_STONITH_DEVICE);
+        }
```

Comment:
Nice type separation like that ...
But on the other hand we could feed is_fence_notification (either beforehand or as part of a combined condition - think we're forbidding the latter in our coding conventions) and avoid some code-duplication.
Not sure which way I really prefer.

Reply:
When touched we could switch to case-sensitive-comparison (iirc case-insensitive is a leftover from the function used previously - right).

Reply:
Definitely on the case sensitivity.

Yeah it would be nice to avoid the duplication but I wanted to leave the possibility of future notification types that might not have the same info.

Reply:
adding an else branch with a new list or moving types between lists shouldn't be any different with the two existing folding into one.